src/llfs/slot_lock_manager.cpp

```diff
@@ -77,16 +77,39 @@ void SlotLockManager::update_upper_bound(slot_offset_type offset)
 //
 StatusOr<SlotReadLock> SlotLockManager::lock_slots(const SlotRange& range, const char* holder)
 {
-  (void)holder;
+  batt::ScopedLock<State> locked{this->state_};
 
-  auto locked = this->state_.lock();
-
-  if (range.lower_bound < this->lower_bound_.get_value()) {
+  if (slot_less_than(range.lower_bound, this->lower_bound_.get_value())) {
     return Status{
         batt::StatusCode::kOutOfRange};  // TODO [tastolfi 2021-10-20]   "the requested value
                                          // extends below the current locked slot range"
   }
 
+  return this->lock_slots_nocheck(locked, range, holder);
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+StatusOr<SlotReadLock> SlotLockManager::lock_all_slots(const char* holder)
+{
+  batt::ScopedLock<State> locked{this->state_};
+
+  const llfs::slot_offset_type lower_bound = this->lower_bound_.get_value();
+
+  return this->lock_slots_nocheck(locked,
+                                  SlotRange{
```

Comment:
Would it be worth having an overload of `lock_slots_nocheck` that takes a `slot_offset` instead of a `SlotRange`? I don't think it's necessary, it just looks strange passing in a `SlotRange` with the same upper and lower bound.

Reply:
I think just passing the slot lower bound would be better if upper bound wasn't used, but it is:

```c++
  this->update_upper_bound_locked(locked, range.upper_bound);
```

Ok to leave as is?

Reply:
ok to leave as is!